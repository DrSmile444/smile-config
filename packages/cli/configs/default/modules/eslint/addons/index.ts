export * from './alias-auto-resolver/alias-auto-resolver.module';
export * from './alias-jsconfig-resolver/alias-jsconfig-resolver.module';
export * from './angular/angular.module';
export * from './node/node.module';
export * from './react/react.module';
export * from './smile-style/eslint-smile-style.module';
export * from './typescript-imports/eslint-typescript-imports.module';
export * from './typescript/eslint-typescript.module';
export * from './vue-typescript/vue-typescript.module';
export * from './vue/vue.module';
