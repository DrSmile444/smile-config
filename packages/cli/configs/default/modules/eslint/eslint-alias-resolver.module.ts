import type {
  AbstractConfigItemModule,
  LintItem,
} from '@smile-config/cli/interfaces';

import { BaseConfigItemModule } from '../../../../src/base';
import {
  EslintAliasAutoResolverModule,
  EslintNodeModule,
  EslintSmileStyleModule,
  EslintTypescriptImportsModule,
  EslintTypescriptModule,
} from './addons';

export class EslintAliasResolverModule
  extends BaseConfigItemModule
  implements AbstractConfigItemModule
{
  title = 'ESLint';
  description = 'Alias Auto Resolver';
  addons = [EslintAliasAutoResolverModule];

  constructor() {
    super(__dirname);
  }
}
