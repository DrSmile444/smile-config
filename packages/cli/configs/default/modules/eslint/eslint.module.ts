import type {
  AbstractConfigItemModule,
  LintItem,
} from '@smile-config/cli/interfaces';

import { BaseConfigItemModule } from '../../../../src/base';
import {
  EslintNodeModule,
  EslintReactModule,
  EslintSmileStyleModule,
  EslintTypescriptImportsModule,
  EslintTypescriptModule,
  EslintVueModule,
} from './addons';

export class EslintModule
  extends BaseConfigItemModule
  implements AbstractConfigItemModule
{
  title = 'ESLint';
  description = 'Enforce JS code style, best practices';
  addons = [
    EslintNodeModule,
    EslintReactModule,
    EslintVueModule,
    EslintTypescriptModule,
    EslintTypescriptImportsModule,
    EslintSmileStyleModule,
  ];

  includeToLintScript: LintItem[] = [
    {
      npmRun: ['lint:js'],
      order: 10,
      additionalCommands: {
        'lint:js': 'eslint . --cache',
        'lint:js:fix': 'npm run lint:js -- --fix',
      },
      when: (packages) => !!packages && !packages['@nrwl/cli'],
      instead: {
        npmRun: ['lint:workspace', 'affected:lint'],
        order: 10,
        additionalCommands: {
          'lint:workspace': 'nx workspace-lint',
          'affected:lint':
            'nx affected:lint --parallel --maxParallel=5 --cache',
          'affected:lint:fix':
            'nx affected:lint --parallel --maxParallel=5 --cache --fix',
        },
      },
    },
  ];

  constructor() {
    super(__dirname);
  }
}
