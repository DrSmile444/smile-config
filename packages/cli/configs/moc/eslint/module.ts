import { ConfigItemModule } from '../../../src/interfaces';
import { EslintTypescriptModule } from './optional/typescript/module';

export class EslintModule implements ConfigItemModule {
  name = 'eslint';

  optional = [
    EslintTypescriptModule,
  ];

  includeToLintScript = [
    {
      command: 'lint:js',
      order: 10,
    }
  ];
}
