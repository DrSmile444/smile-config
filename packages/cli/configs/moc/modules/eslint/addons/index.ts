export * from './typescript/eslint-typescript.module';
