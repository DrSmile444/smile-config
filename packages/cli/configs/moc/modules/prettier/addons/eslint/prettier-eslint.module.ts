import type { AbstractConfigItemModule } from '@smile-config/cli/interfaces';

import { BaseConfigItemModule } from '../../../../../../src/base';

export class PrettierEslintModule
  extends BaseConfigItemModule
  implements AbstractConfigItemModule
{
  title = 'eslint';

  constructor() {
    super(__dirname);
  }
}
