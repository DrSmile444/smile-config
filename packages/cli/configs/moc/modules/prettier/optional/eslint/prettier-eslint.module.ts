import { AbstractConfigItemModule } from '../../../../../src/interfaces';

export class PrettierEslintModule implements AbstractConfigItemModule {
  name = 'eslint';
}
