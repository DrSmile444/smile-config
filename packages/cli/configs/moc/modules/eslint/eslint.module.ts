import type {
  AbstractConfigItemModule,
  LintItem,
} from '@smile-config/cli/interfaces';

import { BaseConfigItemModule } from '../../../../src/base';
import { EslintTypescriptModule } from './addons';

export class EslintModule
  extends BaseConfigItemModule
  implements AbstractConfigItemModule
{
  title = 'eslint';
  addons = [EslintTypescriptModule];
  includeToLintScript: LintItem[] = [
    {
      npmRun: ['lint:js'],
      order: 10,
      additionalCommands: {
        'lint:js': 'eslint . --cache',
        'lint:js:fix': 'npm run lint:js -- --fix',
      },
      when: (packages) => !!packages && !packages['@nrwl/cli'],
      instead: {
        npmRun: ['lint:workspace', 'affected:lint'],
        order: 10,
        additionalCommands: {
          'lint:workspace': 'nx workspace-lint',
          'affected:lint':
            'nx affected:lint --parallel --maxParallel=5 --cache',
          'affected:lint:fix':
            'nx affected:lint --parallel --maxParallel=5 --cache --fix',
        },
      },
    },
  ];

  constructor() {
    super(__dirname);
  }
}
