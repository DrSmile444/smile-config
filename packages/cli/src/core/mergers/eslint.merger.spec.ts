/* eslint-disable @typescript-eslint/no-var-requires,global-require,@typescript-eslint/no-require-imports */
import * as CommentJSON from 'comment-json';
import type { Linter } from 'eslint';
import * as fs from 'fs';

import {
  EslintModule,
  EslintTypescriptModule,
} from '../../../configs/default/modules';
import { EslintMerger } from './eslint.merger';

export const eslintMerger = new EslintMerger();

const readJsonFile = (filePath?: string): Linter.Config | null =>
  filePath
    ? (CommentJSON.parse(fs.readFileSync(filePath).toString()) as Linter.Config)
    : null;

describe('EslintMerger', () => {
  it('should merge mock eslint config', () => {
    const targetA =
      require('./__mocks__/mock-a.eslintrc.json') as Linter.Config;
    const sourceB =
      require('./__mocks__/mock-b.eslintrc.json') as Linter.Config;
    const sourceC =
      require('./__mocks__/mock-c.eslintrc.json') as Linter.Config;
    const expected =
      require('./__mocks__/mock-result.eslintrc.json') as Linter.Config;

    const result1 = eslintMerger.mergeConfigs(targetA, sourceB);
    const result2 = eslintMerger.mergeConfigs(result1, sourceC);

    expect(result1).toMatchSnapshot();
    expect(result2).toEqual(expected);
  });

  it('should merge real eslint config', () => {
    const targetFilePath = readJsonFile(
      new EslintModule().files.find((file) => file.includes('.eslintrc.json'))
    );
    const sourceFilePath = readJsonFile(
      new EslintTypescriptModule().files.find((file) =>
        file.includes('.eslintrc.json')
      )
    );

    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    const result = eslintMerger.mergeConfigs(targetFilePath, sourceFilePath!);
    expect(result).toMatchSnapshot();
  });

  it('should not crash on empty overrides', () => {
    const result = eslintMerger.mergeConfigs(
      { root: true, reportUnusedDisableDirectives: true },
      { root: false, noInlineConfig: true }
    );

    expect(result).toStrictEqual({
      root: false,
      reportUnusedDisableDirectives: true,
      noInlineConfig: true,
    });
  });

  it('should merge plugins', () => {
    const result = eslintMerger.mergeConfigs(
      { plugins: ['eslint'] },
      { plugins: ['smile-config'] }
    );
    expect(result).toStrictEqual({ plugins: ['eslint', 'smile-config'] });
  });
});
