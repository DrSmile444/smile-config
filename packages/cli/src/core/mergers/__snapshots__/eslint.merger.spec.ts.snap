// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EslintMerger should merge mock eslint config 1`] = `
Object {
  "env": Object {
    "commonjs": true,
    "es2021": false,
    "node": true,
  },
  "extends": Array [
    "../../.eslintrc.json",
    "rule-b-extends",
  ],
  "ignorePatterns": Array [
    "!**/*",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
        "*.tsx",
        "*.js",
        "*.jsx",
      ],
      "plugins": Array [
        "rule-a-plugin",
        "rule-b-plugin",
      ],
      "rules": Object {
        "rule-a": "replaced-by-b",
        "rule-a-2": "origin",
        "rule-b": "origin",
      },
    },
    Object {
      "files": Array [
        "*.ts",
        "*.tsx",
      ],
      "plugins": Array [
        "rule-b-ts-plugin",
      ],
      "rules": Object {
        "rule-b-ts": "origin",
      },
    },
  ],
}
`;

exports[`EslintMerger should merge real eslint config 1`] = `
Object {
  "env": Object {
    "es2021": true,
  },
  "extends": Array [],
  "overrides": Array [
    Object {
      "extends": Array [
        "plugin:import/typescript",
        "plugin:@typescript-eslint/all",
      ],
      "files": Array [
        "*.ts",
        "*.tsx",
      ],
      "parserOptions": Object {
        "createDefaultProgram": true,
        "project": Array [
          "tsconfig.json",
        ],
      },
      "rules": Object {
        "@typescript-eslint/array-type": Array [
          "error",
          Object {
            "default": "array",
          },
        ],
        "@typescript-eslint/consistent-type-definitions": "error",
        "@typescript-eslint/dot-notation": "off",
        "@typescript-eslint/explicit-member-accessibility": Array [
          "off",
          Object {
            "accessibility": "explicit",
          },
        ],
        "@typescript-eslint/indent": Array [
          "error",
          2,
        ],
        "@typescript-eslint/lines-between-class-members": Array [
          "error",
          "always",
          Object {
            "exceptAfterSingleLine": true,
          },
        ],
        "@typescript-eslint/member-delimiter-style": Array [
          "error",
          Object {
            "multiline": Object {
              "delimiter": "semi",
              "requireLast": true,
            },
            "singleline": Object {
              "delimiter": "semi",
              "requireLast": false,
            },
          },
        ],
        "@typescript-eslint/member-ordering": "error",
        "@typescript-eslint/naming-convention": Array [
          "error",
          Object {
            "custom": Object {
              "match": false,
              "regex": "^I[A-Z]",
            },
            "format": Array [
              "PascalCase",
            ],
            "selector": "interface",
          },
          Object {
            "format": Array [
              "strictCamelCase",
              "snake_case",
              "UPPER_CASE",
            ],
            "selector": "property",
          },
        ],
        "@typescript-eslint/no-duplicate-imports": Array [
          "error",
        ],
        "@typescript-eslint/no-empty-function": "off",
        "@typescript-eslint/no-empty-interface": "error",
        "@typescript-eslint/no-misused-new": "error",
        "@typescript-eslint/no-non-null-assertion": "error",
        "@typescript-eslint/no-shadow": "error",
        "@typescript-eslint/no-type-alias": "off",
        "@typescript-eslint/no-unused-expressions": "error",
        "@typescript-eslint/no-use-before-define": "error",
        "@typescript-eslint/prefer-function-type": "error",
        "@typescript-eslint/prefer-readonly-parameter-types": "off",
        "@typescript-eslint/quotes": Array [
          "error",
          "single",
        ],
        "@typescript-eslint/semi": Array [
          "error",
          "always",
        ],
        "@typescript-eslint/strict-boolean-expressions": "off",
        "@typescript-eslint/type-annotation-spacing": "error",
        "@typescript-eslint/unified-signatures": "error",
        "indent": "off",
        "indent-legacy": "off",
        "no-duplicate-imports": "off",
        "no-shadow": "off",
      },
    },
  ],
  "root": true,
  "settings": Object {
    "import/resolver": Object {
      "node": Object {
        "extensions": Array [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
      },
      "typescript": Object {},
    },
  },
}
`;
